src/epiviz.gl/webgl-vis.js

```diff
@@ -108,33 +108,26 @@ class WebGLVis {
         if (message.data.closestPoint === undefined) {
           return;
         }
-        this.parent.dispatchEvent(
-          new CustomEvent("pointHovered", { detail: message })
-        );
+        this.dispatchEvent("pointHovered", message);
       } else if (message.data.type === "getClickPoint") {
         if (message.data.closestPoint === undefined) {
           return;
         }
-        this.parent.dispatchEvent(
-          new CustomEvent("pointClicked", { detail: message })
-        );
+        this.dispatchEvent("pointClicked", message);
       } else if (
         message.data.type === "selectBox" ||
         message.data.type === "selectLasso"
       ) {
-        this.parent.dispatchEvent(
-          new CustomEvent("onSelectionEnd", { detail: message })
-        );
+        this.dispatchEvent("onSelectionEnd", message);
         this.dataWorkerStream.push(message);
-        console.log(this.dataWorkerStream);
       }
     };
     this.dataWorker.onerror = (e) => {
       throw e;
     };
 
     // Needs to be called at the end of addToDOM so mouseReader has correct dimensions to work with
-    this.mouseReader.init();
+    this.mouseReader.init(this.parent);
```

Comment:
L82 in `mouse-reader.js` has no params in function declaration but you seem to be passing this in?

Reply:
Ah, I refactored the code and this one got missed by my eyes. Thanks for pointing this out.